options/posix/generic/sys-utsname.cpp

```diff
@@ -0,0 +1,24 @@
+
+#include <string.h>
+#include <sys/utsname.h>
+#include <errno.h>
+
+#include <bits/ensure.h>
+#include <mlibc/debug.hpp>
+#include <internal-config.h>
+#include <mlibc/posix-sysdeps.hpp>
+
+int uname(struct utsname *p) {
+	__ensure(p);
```

Comment:
Instead of this ``__ensure`` we should just set ``errno`` to ``EINVAL`` and return -1. This is also what glibc does.